PhotoBooth/CountdownIndicatorView.swift

```diff
@@ -0,0 +1,99 @@
+import UIKit
+
+class CountdownIndicatorView: UIView {
+    private let viewModel: CountdownIndicatorViewModeling
+
+    private let pulseLayer = CAShapeLayer()
+    private let startAngle = -CGFloat.pi / 2
+
+    lazy var countdownLabel: UILabel = {
+        let label = UILabel()
+        label.text = viewModel.timerLabelText
+        label.font = UIFont.boldFont(size: 23)
+        label.textAlignment = .center
+        label.textColor = .white
+        label.backgroundColor = UIColor.black.withAlphaComponent(0.8)
+        label.numberOfLines = 0
+        label.adjustsFontSizeToFitWidth = true
+        label.layer.cornerRadius = 20
+        label.layer.masksToBounds = true
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    init(viewModel: CountdownIndicatorViewModeling) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        commonInit()
+    }
+
+    func updateWith(timeRemaining: Seconds, animated: Bool) {
+        if animated == true {
+            viewModel.update(timeRemaining: timeRemaining)
+            countdownLabel.text = viewModel.timerLabelText
```

Comment:
You're doing both 36 and 37 in both cases - that should be an indicator to pull it out:

```swift 
    viewModel.update(timeRemaining: timeRemaining)
    countdownLabel.text = viewModel.timerLabelText
    
    if animated { animatePulseLayer() }
```